File: Programowanie_Obiektowe/PO_JAVA/src/Lis.java
public class Lis extends Zwierze
{
    public Lis(int x, int y, Swiat swiat)
    {
        super(Stale.LIS_SILA, Stale.LIS_INICJATYWA, x, y, swiat);
    }

    @Override
    public void akcja()
    {
        wiek++;

        int[] XY = swiat.wylosujSasiada(this);
        if (XY != null)
        {
            Organizm cel = swiat.getOrganizmNa(XY[0], XY[1]);
            if (cel == null)        //przesunieto zwierze
            {
                swiat.dodajLog(rysowanie() + " (" + x + ", " + y + ", " + sila + ") przesuwa się na: " + XY[0] + ", " + XY[1]);
                swiat.ustawOrganizmNa(x, y, null);
                x = XY[0];
                y = XY[1];
                swiat.ustawOrganizmNa(x, y, this);
            }
            else                    //zwierze zaatakowalo
            {
                if (cel.getSila() > sila)   //lis wyczul zagrozenie
                {
                    swiat.dodajLog(rysowanie() + " (" + x + ", " + y + ", " + sila + ") wyczuł zagrożenie");
                }
                else
                {
                    cel.kolizja(this);
                }
            }
        }
        else
        {
            swiat.dodajLog(rysowanie() + " (" + x + ", " + y + ", " + sila + ") brak dostepnych ruchow");
        }
    }

    @Override
    public char rysowanie()
    {
        return Stale.LIS;
    }

    @Override
    public String nazwa()
    {
        return Stale.LIS_S;
    }

    @Override
    public Organizm stworzPotomka(int x, int y)
    {
        return new Lis(x, y, swiat);
    }
}

File: Programowanie_Obiektowe/PO_C/PO_C/Lis.cpp
#include "Lis.h"

Lis::Lis(int x, int y, Swiat& swiat) : Zwierze(LIS_SILA, LIS_INICJATYWA, x, y, swiat) {}

void Lis::akcja()
{
    wiek++;

    int dx[8] = { -1, 0, 1, -1, 1, -1, 0, 1 };
    int dy[8] = { 1, 1, 1, 0, 0, -1, -1, -1 };

    while (true)
    {
        int dir = rand() % 8;

        int newX = x + dx[dir];
        int newY = y + dy[dir];

        if (swiat.czyNaPlanszy(newX, newY))
        {
            Organizm* cel = swiat.getOrganizmNa(newX, newY);
            if (cel == nullptr)         //przesunieto zwierze
            {
                swiat.setLog(swiat.getLog() + string(1, rysowanie()) + " (" + to_string(x) + ", " + to_string(y) + ", " + to_string(sila) + ") przesuwa sie na: " + to_string(newX) + ", " + to_string(newY) + "\n");

                swiat.ustawOrganizmNa(y, x, nullptr);
                x = newX;
                y = newY;
                swiat.ustawOrganizmNa(y, x, this);
            }
            else                        //zwierze zaatakowalo
            {
                if (cel->getSila() > sila)  //lis wyczul zagrozenie
                {
                    swiat.setLog(swiat.getLog() + string(1, rysowanie()) + " (" + to_string(x) + ", " + to_string(y) + ", " + to_string(sila) + ") wyczul zagrozenie" + "\n");
                }
                else
                {
                    cel->kolizja(*this);
                }
            }
            break;
        }
    }
}

char Lis::rysowanie() const
{
    return LIS;
}

Organizm* Lis::stworzPotomka(int x, int y) const
{
    return new Lis(x, y, swiat);
}